types/node/worker_threads.d.ts

```diff
@@ -60,13 +60,21 @@ declare module "worker_threads" {
         stdout?: boolean;
         stderr?: boolean;
         execArgv?: string[];
+        resourceLimits?: ResourceLimits;
+    }
+
+    interface ResourceLimits {
+        maxYoungGenerationSizeMb: number;
+        maxOldGenerationSizeMb: number;
+        codeRangeSizeMb: number;
     }
 
     class Worker extends EventEmitter {
         readonly stdin: Writable | null;
         readonly stdout: Readable;
         readonly stderr: Readable;
         readonly threadId: number;
+        readonly resourceLimits?: ResourceLimits;
```

Comment:
I think `resourceLimits` is always present but each property may be missing (you get `{}` if no limit was given).
Therefore change `ResourceLimits ` also to
```
interface ResourceLimits {
    maxYoungGenerationSizeMb?: number;
    maxOldGenerationSizeMb?: number;
    codeRangeSizeMb?: number;
}
```
Once again this fits fine to options as the whole `resourceLimits` property is optional but if it is present each limit property is optional.

Reply:
So it would be `readonly resourceLimits: Partial<ResourceLimits>;`?

Reply:
no, `ResourceLimits` should be changed to all properties optional. This matches in both cases.